api/http/middlewares/admin_monitor.go

```diff
@@ -0,0 +1,62 @@
+package middlewares
+
+import (
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+
+	httperror "github.com/portainer/libhttp/error"
+)
+
+const RedirectReasonAdminInitTimeout string = "AdminInitTimeout"
+
+// AdminMonitor is an entity used to maintain the administrator initialization status
+type AdminMonitor struct {
+	lock              sync.RWMutex
+	adminInitDisabled bool
+}
+
+//  NewAdminMonitor creates a new gate wrapper
+func NewAdminMonitor(timeoutSignal <-chan interface{}) *AdminMonitor {
```

Comment:
IMO the whole functionality can be move into the adminmonitor service, which can either expose the middleware or change the middleware to take the adminmonitor service (or just a func that returns if it's disabled)

Reply:
fixed